parser/prism/Translator.cc

```diff
@@ -1193,27 +1207,45 @@ std::unique_ptr<parser::Node> Translator::translateStatements(pm_statements_node
     return make_unique<parser::Begin>(parser.translateLocation(&stmtsNode->base.location), std::move(sorbetStmts));
 }
 
-std::unique_ptr<parser::Node> Translator::translateConstantPath(pm_constant_path_node *node, bool isAssignment) {
-    pm_location_t *loc = &node->base.location;
+// Handles any one of the Prism nodes that models any kind of assignment to a constant or constant path.
+template <typename PrismLhsNode, typename SorbetLHSNode>
+unique_ptr<SorbetLHSNode> Translator::translateConst(PrismLhsNode *node) {
+    static_assert(std::is_same_v<SorbetLHSNode, parser::Const> || std::is_same_v<SorbetLHSNode, parser::ConstLhs>,
+                  "Invalid LHS type. Must be one of `parser::Const` or `parser::ConstLhs`.");
 
-    std::string_view name = parser.resolveConstant(node->name);
+    auto constexpr isConstantPath = std::is_same_v<PrismLhsNode, pm_constant_path_target_node> ||
+                                    std::is_same_v<PrismLhsNode, pm_constant_path_write_node> ||
+                                    std::is_same_v<PrismLhsNode, pm_constant_path_node>;
 
     std::unique_ptr<parser::Node> parent;
-    if (node->parent) {
-        // This constant reference is chained onto another constant reference.
-        // E.g. if `node` is pointing to `B`, then then `A` is the `parent` in `A::B::C`.
-        parent = translate(node->parent);
-    } else {                                                // This is a fully qualified constant reference, like `::A`.
-        pm_location_t *delimiterLoc = &node->delimiter_loc; // The location of the `::`
-        parent = make_unique<parser::Cbase>(parser.translateLocation(delimiterLoc));
```

Comment:
The diff is noisy here, but this is pretty much the same content, just indented into the new `if constexpr` body